scipy/signal/ltisys.py

```diff
@@ -500,6 +501,419 @@ def freqresp(self, w=None, n=10000):
+
+    """
+    def __new__(cls, *system):
+        """Handle object conversion if input is an instance of lti"""
+        if len(system) == 1:
+            if isinstance(system[0], ZerosPolesGain):
+                return copy.deepcopy(system[0])
+            if isinstance(system[0], lti):
+                return system[0].to_zpk()
+
+        # No special conversion needed
+        return super(ZerosPolesGain, cls).__new__(cls)
+
+    def __init__(self, *system):
+        """Initialize the zeros, poles, gain LTI system
+
+        Parameters
+        ----------
+        system : arguments
+            The following arguments are possible
+            * an instance of the lti class (StateSpace, TransferFunction,
+              ZerosPolesGain)
+            * (zeros, poles, gain)
+
+        """
+        # Conversion of lti instances is handled in __new__
+        if isinstance(system[0], lti):
+            return
+
+        super(ZerosPolesGain, self).__init__(self, *system)
+
+        self._zeros = None
+        self._poles = None
+        self._gain = None
+
+        self.zeros, self.poles, self.gain = system
+
+    def __repr__(self):
+        """Return representation of the ZerosPolesGain system"""
+        return '{0}(\n{1},\n{2},\n{3}\n)'.format(
+            self.__class__.__name__,
+            repr(self.zeros),
+            repr(self.poles),
+            repr(self.gain),
+            )
+
+    @property
+    def zeros(self):
+        return self._zeros
+
+    @zeros.setter
+    def zeros(self, zeros):
+        self._zeros = atleast_1d(zeros)
+
+        # Update dimensions
+        if len(self.zeros.shape) > 1:
+            self.outputs, self.inputs = self.zeros.shape
+        else:
+            self.outputs = 1
+            self.inputs = 1
+
+    @property
+    def poles(self):
+        return self._poles
+
+    @poles.setter
+    def poles(self, poles):
+        self._poles = atleast_1d(poles)
+
+    @property
+    def gain(self):
+        return self._gain
+
+    @gain.setter
+    def gain(self, gain):
+        self._gain = gain
+
+    def _copy(self, system):
+        """Copy the parameters of another ZerosPolesGain system
+
+        Parameters
+        ----------
+        system : instance of ZerosPolesGain
+            The zeros, poles gain system that is to be copied
+
+        """
+        self.poles = system.poles
+        self.zeros = system.zeros
+        self.gain = system.gain
+
+    def to_tf(self):
+        """Convert system representation to transfer function.
+
+        Returns
+        -------
+        sys : instance of TransferFunction
+            Transfer function of the current system
+
+        """
+        return TransferFunction(*zpk2tf(self.zeros, self.poles, self.gain))
+
+    def to_zpk(self):
+        """Convert system representation to zeros, poles, gain.
+
+        Returns
+        -------
+        sys : instance of ZerosPolesGain
+            The current system (self)
+
+        """
+        return self
```

Comment:
Alright, I'll change ```to_ss()```and the likes to copy. But general inputs such as state space matrices are still refs. That way at least when you do the conversion you're sure there are no hidden refs. Thanks for the feedback!